pkg/instance/service.go

```diff
@@ -313,6 +316,71 @@ func (s service) deployDeploymentInstance(token string, instance *model.Deployme
 	return nil
 }
 
+func (s service) DeleteDeployment(id uint) error {
+	deployment, err := s.FindDeploymentById(id)
+	if err != nil {
+		return err
+	}
+
+	deploymentGraph, err := s.validateNoCycles(deployment.Instances)
+	if err != nil {
+		return err
+	}
+
+	instances, err := deploymentOrder(deployment, deploymentGraph)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Please remove the newline as `deploymentOrder` and `Reverse` are closely related. Or best replace it with a comment of why you are calling `Reverse`. This might not be obvious at first.